qos/evm/response_generic.go

```diff
@@ -54,9 +54,8 @@ func (r responseGeneric) GetObservation() qosobservations.EVMEndpointObservation
 	return qosobservations.EVMEndpointObservation{
 		ResponseObservation: &qosobservations.EVMEndpointObservation_UnrecognizedResponse{
 			UnrecognizedResponse: &qosobservations.EVMUnrecognizedResponse{
-				JsonrpcResponse: &qosobservations.JsonRpcResponse{
-					Id: r.jsonRPCResponse.ID.String(),
-				},
+				// Include JSONRPC response's details in the observation.
+				JsonrpcResponse:         r.jsonRPCResponse.GetObservation(),
```

Comment:
Why is this not `JsonrpcResponse: r.jsonRPCResponse`?

Reply:
It is populating the `JsonrpcResponse` field of the observation (which intentionally does not include an observation postfix, as it is inside an Observation struct.
Please reopen if this needs to be revisited.